packages/openmrs/src/Utils.js

```diff
@@ -18,20 +17,45 @@ export const prepareNextState = (state, response, callback = s => s) => {
 export async function fetchAndLog(method, url, opts) {
   try {
     const response = await commonRequest(method, url, opts);
-    logResponse(response);
+    const { statusCode, duration } = response;
+
+    const urlWithQuery = Object.keys(opts.query || {}).length
+      ? `${url}?${new URLSearchParams(opts.query).toString()}`
+      : url;
+
+    const message = `${method} ${urlWithQuery} - ${statusCode} in ${duration}ms`;
+    if (response instanceof Error) {
+      console.error(message);
+    } else {
+      console.log(message);
+    }
+    response.url = urlWithQuery;
     return response;
   } catch (err) {
     return err;
   }
 }
 
+// TODO get() should pass autopage true and options
+// TODO http.*() functions should all pass autopage false (at least for now)
+// TODO this currently returns { result, response }. Is that right? It'll break everything!!
 export async function request(state, method, path, options = {}) {
+  const { username, password, instanceUrl: baseUrl } = state.configuration;
+
   const {
-    baseUrl = state.configuration.instanceUrl,
+    // baseUrl = instanceUrl, // TODO should not be able to override baseUrl on config
```

Comment:
I've left this commented out as a reminder: we need to check the implications of removing this line